src/ember-app.js

```diff
@@ -235,7 +239,7 @@ class EmberApp {
       })
       .then(() => result.instanceBooted = true)
       .then(() => instance.visit(path, bootOptions))
-      .then(() => waitForApp(instance))
+      .then(() => fastbootInfo.deferredPromise)
```

Comment:
What if during the `visit`, the instance mutated the `fastbootInfo` object? Can you make sure this isn't a stale copy?